### DISCUSSION TITLE
how can I run these github-maintained Runner Images locally?

### DISCUSSION THREAD

**Discussion by jtmoon79** [community/community#160647] [Actions]
I want to test a few things within a github-maintained Runner Image. What would be the docker command to run a Runner Image locally? (what is the docker repository URL)

For example, I'd like to run the Runner Image for Ubuntu 22 tag [`20230409.1`](https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230409.1). What would be the command?

**Comment by v-sergei-pyshnoi** [comment]
Hello @jtmoon79 . Runner images that using in Actions it`s a not docker container. It is a Virtual Machine. You can see [in this doc](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md) how exactly is image builds.

**Reply by romanr** [reply]
Anyone knows where to find instruction how to run an image?

**Reply by pkit** [reply]
Run in where?

**Reply by vpolikarpov-akvelon** [reply]
@romanr, there is [a small section](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md#generated-machine-deployment) in the manual about running generated image in your Azure subscription.

**Comment by pkit** [comment]
@jtmoon79 these are VM images, you can run them with KVM. But not with docker.

**Reply by ssbarnea** [reply]
I was looking exactly into that but I see no links about how to download these images for local use. Does anyone have any links/scripts for doing so?

That would be very useful as github maintained runner images do diverge considerably from pure OS installs and often we face bugs that happen only on CI.

**Reply by pkit** [reply]
First, you need to build these images (yes, can be built only on Azure, you will need to have an Azure account): https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md#manual-image-generation
Then you can use the resulting image files in KVM or virt-manager.

**Reply by StephenHodgson** [reply]
Would be great if the images could be pushed to a docker container image we can pull locally to setup self hosted runners

**Reply by iZnoGod722** [reply]
These are actually managed images used to spin up virtual machines, not container based images.
